### ISSUE TITLE
Decompiler: raise capturing lambdas over <>c__DisplayClass delegate creations (EVIL CS1001 cluster)

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#2945]
## Summary

The first full **EVIL** corpus benchmark run (12,000 difficulty-ranked hard-IL
methods, all 26 source-resolvable assemblies matched) surfaces a single dominant
genuine-decompiler defect family: **capturing lambdas are not raised when their
closure is a `<>c__DisplayClassN_M` instance referenced through a
`new Func<…>(local.<Owner>b__N)` delegate creation.** The raiser leaves the raw
compiler-generated method reference and the printer emits the angle-bracketed
`<Owner>b__N` name verbatim, producing invalid C# (`CS1001: Identifier
expected`). This is the largest single reason bucket in the run (780 CS1001 rows).

## EVIL benchmark measurement

Full honest run (`--benchmark-authored-corpus evil/corpus.jsonl`, 26/26
assemblies, 12,000 targets evaluated):

| Bucket | Count |
| --- | --- |
| Correct (valid, matches authored) | 1,501 |
| Valid but differs | 5,290 (frontier IL-exact 3,097 / IL-diff 2,181) |
| Invalid (does not round-trip) | 5,209 |

Of the 5,209 Invalid, ~3,163 are reference/accessibility codes
(`CS0234`/`CS0246`/`CS1069`/`CS0122`…) attributable to compile-back **shell
reference gaps** on the broad third-party dependency surface (the decompiler
emits the correct type; the RTS shell just lacks the reference), **not**
decompiler-emission bugs. The genuine decompiler-invalid signal is ~971
syntax-malformation rows, led by **`CS1001` = 780 ("closure-stalled")**.

## Root cause (reproduced)

`System.CommandLine.Command.SetAction#0` decompiles to:

```csharp
___c__DisplayClass30_0 V_0 = new ___c__DisplayClass30_0();
V_0.action = action;
if (V_0.action is null)
    throw new ArgumentNullException("action");
Action = new AnonymousSynchronousCommandLineAction(new Func<ParseResult, int>(V_0.<SetAction>b__0));
```

`V_0.<SetAction>b__0` is a raw reference to the display-class instance method,
emitted with the illegal `<…>` identifier → `CS1001`. Note the display-class
*type* is sanitized (`<>c__DisplayClass30_0` → `___c__DisplayClass30_0`) but the
*method* reference is not — and even if it were sanitized, it would reference a
private compiler-generated method that does not exist in valid C#. The correct
output folds the delegate into a capturing lambda, e.g.:

```csharp
Action = new AnonymousSynchronousCommandLineAction(parseResult => { /* body of b__0 */ });
```

**The gap is specifically capturing lambdas.** Non-capturing lambdas already
raise correctly in the same corpus — e.g. `HelpBuilder.WriteColumns` emits
`items.Select(x => x.FirstColumnText.Length)` fine, but stalls on the capturing
`new Func<…>(V_3.<WriteColumns>b__3)` in the same method. A related but distinct
illegal-identifier source appears in local-function iterator state-machine
initializers (`{ <>3__first = first }`).

### Reproductions (repo-buildable assembly)

All from `System.CommandLine` (`3.0.0-preview.5`):

- `System.CommandLine.Command::SetAction#0`
- `System.CommandLine.CompletionSourceExtensions::Add#0`
- `System.CommandLine.Command::GetCompletions#0`
- `System.CommandLine.Help.HelpBuilder::WriteColumns#0`

109 closure-stalled `CS1001` rows appear in the 3-assembly subset
(`System.CommandLine` + `FluentValidation` + `FluentValidation.DependencyInjectionExtensions`)
alone.

## Fix location

The fix belongs entirely in the **decompiler** — extend capturing-lambda raise
to fold `new Func<…>(displayClassLocal.<Owner>b__N)` delegate creations into
lambda expressions over the display-class fields (captured locals), and ensure
the printer never emits an un-raised `<…>b__N` identifier.

**Do not compensate in RTS.** RTS is the compile-back harness; giving it C#
knowledge to rewrite these references would hide the product gap and violate the
harness boundary (AGENTS.md → *Harness boundary*).

Tracks toward the quality ladder (#1599): a measured invalid-`Full` gap with a
fixture/corpus signal and a concrete success bar (these rows flip Invalid →
Valid/Correct).

---

## Package versions (for reproduction)

Reproduced examples in this issue are **System.CommandLine 3.0.0-preview.5.26302.115** (`net10.0`, real-world pin). The cluster (~780 rows) is general across the corpus; representative packages:

- **Moq 4.20.72** (97 rows)
- **Hangfire.Core 1.8.24** (94 rows)
- **FluentValidation 12.1.1** (87 rows)

**Provenance is retained durably**: every row in `vendor/authored-source-corpus:evil/corpus.jsonl` carries `assembly`, `assemblyVersion`, `tfm` (the NuGet **package version** for top-N sweep packages, e.g. `2.1.79`; a real **TFM** like `net10.0`/`net9.0` for the 14 real-world pins), plus `sourceUrl` (pinned source commit) and `checksum` (SHA256). `eng/prepare-evil-corpus.sh` also emits `<pool>/sweep-manifest.json` with resolved versions/TFMs.

---

## Demonstrate with `dotnet-inspect` (product commands, no harness)

```bash
dnx dotnet-inspect -y -- member "FluentValidation.AbstractValidator`1" RuleSet --package FluentValidation@12.1.1 --all -S "Decompiled Source"
```

The capturing lambdas over the `<>c__DisplayClassN_M` closure are emitted as method-group references to the compiler-generated method, producing an **unspellable** identifier (CS1001):

```csharp
Func<string, string> S_2 = new Func<string, string>(___c.___9.<RuleSet>b__23_0);
using (IDisposable V_1 = Rules.OnItemAdded(new Action<IValidationRuleInternal<T>>(V_0.<RuleSet>b__1)))
```

`V_0.<RuleSet>b__1` / `___c.___9.<RuleSet>b__23_0` are the un-raised capturing lambda bodies. `-S "Annotated Source"` shows the underlying `ldftn <RuleSet>b__1` + `newobj Action::.ctor`. Fix in the decompiler raise/printer (capturing-lambda closure raise), **not** RTS.
